src/bson.rs

```diff
@@ -529,7 +529,7 @@ impl Bson {
     /// This function mainly used for [extended JSON format](https://docs.mongodb.com/manual/reference/mongodb-extended-json/).
     // TODO RUST-426: Investigate either removing this from the serde implementation or unifying
     // with the extended JSON implementation.
-    pub(crate) fn into_extended_document(self) -> Document {
+    pub(crate) fn into_extended_document(self, rawbson: bool) -> Document {
```

Comment:
rather than implementing various `MapAccess` structs, the `Bson` deserializer instead converts things into their "extended document" form and then uses a single `MapAccess` implementation to walk over the resultant document. The changes here modify this method to create the documents in the form that the raw BSON types expect where necessary.